src/v/config/node_config.cc

```diff
@@ -120,14 +120,50 @@ node_config::node_config() noexcept
       {.visibility = visibility::user},
       {}) {}
 
+void validate_multi_node_property_config(
+  std::map<ss::sstring, ss::sstring>& errors) {
+    auto const& cfg = config::node();
+    const auto& kafka_api = cfg.kafka_api.value();
+    for (auto const& ep : kafka_api) {
+        const auto& n = ep.name;
+        auto authn_method = ep.authn_method;
+        if (authn_method.has_value()) {
+            if (authn_method == config::broker_authn_method::mtls_identity) {
+                auto kafka_api_tls = config::node().kafka_api_tls();
+                auto tls_it = std::find_if(
+                  kafka_api_tls.begin(),
+                  kafka_api_tls.end(),
+                  [&n](const config::endpoint_tls_config& ep) {
+                      return ep.name == n;
+                  });
+                if (
+                  tls_it == kafka_api_tls.end() || !tls_it->config.is_enabled()
+                  || !tls_it->config.get_require_client_auth()) {
+                    errors.emplace(
+                      "kafka_api.authentication_method",
+                      ssx::sformat(
+                        "kafka_api {} configured with {}, but there is not an "
+                        "equivalent kafka_api_tls config that requires client "
+                        "auth. Setting authentication_method to {}",
```

Comment:
If node_config returns errors, we throw from application::hydrate_config -- so that last sentence here can go: we aren't going to use a none auth method, we'll just not run.

Reply:
Done